### ISSUE TITLE
SD Card will not be tested in `disk/read_preformance_.*`

### ISSUE THREAD

**Issue by LiaoU3** [canonical/checkbox#1272]
### Bug Description

If a DUT has SD Card installed with system, the job [disk/read_performance_.*](https://github.com/canonical/checkbox/blob/c894f519ee728ab7bb34b10d39387bd9bd57e7d4/providers/base/units/disk/jobs.pxu#L70C5-L70C27) will not be run on the SD Card.

The root cause is because the [template-filter](https://github.com/canonical/checkbox/blob/c894f519ee728ab7bb34b10d39387bd9bd57e7d4/providers/base/units/disk/jobs.pxu#L67) require `device.category == 'DISK'`, however, the output from [device](https://github.com/canonical/checkbox/blob/c894f519ee728ab7bb34b10d39387bd9bd57e7d4/providers/resource/jobs/resource.pxu#L125), it says the category of SD Card is `CARDREADER` as below.
```
path: /devices/platform/axi/ff9d0000.usb/fe200000.usb/xhci-hcd.1.auto/usb1/1-1/1-1.1/1-1.1:1.0/host0/target0:0:0/0:0:0:0
name: sda
bus: scsi
category: CARDREADER
driver: sd
product: Ultra HS-COMBO
vendor: Generic
product_slug: Ultra_HS-COMBO
vendor_slug: Generic
```

Submission: https://certification.canonical.com/hardware/202202-29984/submission/371552/

### To Reproduce

1. Prepare a machine with SD Card installed
2. checkbox.checkbox-cli list-bootstrapped com.canonical.certification::disk-cert-automated

### Environment

 - Checkbox Type: Snap
 - Checkbox Version: 4.0.0-dev264

### Relevant log output

```shell
ubuntu@kria:~$ checkbox-limerick-classic.checkbox-cli list-bootstrapped com.canonical.certification::disk-cert-automated
$PROVIDERPATH is defined, so following provider sources are ignored ['/snap/checkbox-limerick-classic/x1/providers/checkbox-provider-limerick', '/home/ubuntu/.local/share/plainbox-providers-1', '/var/tmp/checkbox-providers-develop'] 
Skipped file: /snap/checkbox24/current/providers/checkbox-provider-base/units/stress/suspend_cycles_reboot.md
com.canonical.certification::executable
com.canonical.certification::disk/detect
com.canonical.plainbox::manifest
com.canonical.certification::disk/check-software-raid


ubuntu@kria:~$ lsblk 
NAME       MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0        7:0    0  36.2M  1 loop /snap/bugit/44
loop1        7:1    0 311.3M  1 loop /snap/checkbox24/15
loop2        7:2    0   1.9M  1 loop /snap/checkbox-limerick-classic/x1
loop3        7:3    0  59.8M  1 loop /snap/core20/2321
loop4        7:4    0  69.2M  1 loop /snap/core22/1383
loop5        7:5    0  61.2M  1 loop /snap/core24/410
loop6        7:6    0  61.2M  1 loop /snap/core24/424
loop7        7:7    0  33.7M  1 loop /snap/snapd/21467
loop8        7:8    0  33.7M  1 loop /snap/snapd/21761
sda          8:0    1  14.8G  0 disk 
├─sda1       8:1    1     1G  0 part /boot/firmware
└─sda2       8:2    1  13.8G  0 part /
sdb          8:16   1  28.9G  0 disk 
└─sdb1       8:17   1  28.9G  0 part 
mtdblock0   31:0    0   512K  1 disk 
mtdblock1   31:1    0   512K  1 disk 
mtdblock2   31:2    0   128K  0 disk 
mtdblock3   31:3    0   128K  0 disk 
mtdblock4   31:4    0   768K  0 disk 
mtdblock5   31:5    0    13M  0 disk 
mtdblock6   31:6    0   512K  1 disk 
mtdblock7   31:7    0    13M  0 disk 
mtdblock8   31:8    0   512K  1 disk 
mtdblock9   31:9    0     1M  0 disk 
mtdblock10  31:10   0     2M  1 disk 
mtdblock11  31:11   0     2M  1 disk 
mtdblock12  31:12   0   128K  0 disk 
mtdblock13  31:13   0   128K  0 disk 
mtdblock14  31:14   0   256K  1 disk 
mtdblock15  31:15   0   128K  0 disk 
mtdblock16  31:16   0  29.4M  0 disk
```

**Comment by LiaoU3** [comment]
We have the below machine avaible on testfliger, you could try on this.
https://certification.canonical.com/hardware/202307-31906/

**Comment by fernando79513** [comment]
Hello! I was reviewing the issue and I have some questions on it.

- It is straightforward to add the device.category == 'CARDREADER' to the disk performance test, but do we want to run this test for all devices with a SD card, even if that card is not a card with the system installed?
- Do we want to run other disk tests for the SD card (disk/stats_., disk/storage_device_., etc) or just disk/read_performance_*

**Comment by LiaoU3** [comment]
> Hello! I was reviewing the issue and I have some questions on it.
> 
>     * It is straightforward to add the device.category == 'CARDREADER' to the disk performance test, but do we want to run this test for all devices with a SD card, even if that card is not a card with the system installed?
> 
>     * Do we want to run other disk tests for the SD card (disk/stats_., disk/storage_device_., etc) or just disk/read_performance_*

1. From my point of view, we should only run tests on the SD card which has system installed.
2. I think all of the tests should be run (disk/stats_., disk/storage_device_., etc)

**Comment by LiaoU3** [comment]
> Also, when running the `list-bootstrap` command after adding the `bootstrap_include`, I noticed that there are 16 "mtdblock" partitions, some of them with less than 1MB, that are included in the tests:
> 
> ```
> mtdblock0   31:0    0   512K  1 disk 
> mtdblock1   31:1    0   512K  1 disk 
> ...
> mtdblock16  31:16   0  29.4M  0 disk
> ```
> 
> These are failing the performance tests because they are too small, should we ingnore all `mtdblock` partitions on the tests? Or maybe skip the size is too small?

I think skip the size is too small will be better, since we are not sure if anyone want to run it.

**Comment by fernando79513** [comment]
>I found that ignoring the SD devices for storage tests may actually be an intended behavior, not a bug
https://github.com/canonical/checkbox/blob/main/checkbox-support/checkbox_support/parsers/udevadm.py#L450-L466
```
        if self.driver:
...
            # Only consider eMMC internal drives as DISK.
            # As it's not possible to distinguish them from simple MMC
            # removable storage, only those with a partition mounted as /
            # will be considered.
            # To avoid categorizing SD cards as main storage and run heavy
            # tests on them (e.g on a pandaboard) only devices with the udev
            # property MMC_TYPE == MMC are accepted.
            if self.driver.startswith("mmc"):
                if self._mmc_type == "MMC" and find_pkname_is_root_mountpoint(
                    self.name, self._lsblk
                ):
                    return "DISK"
                else:
                    return "CARDREADER"
```
If SD cards are where system is installed, I think running tests on those SD cards are reasonable. Otherwise, there will not be any disk tests on it.

**Comment by zongminl** [comment]
Fixed by #1451
